### PR TITLE
Allow users to toggle --only-binary flag

### LINKED ISSUE #8: `--only-binary :all:` creates unresolvable package specs
When using the setup-uv action, `--no-binary :all:` causes the package installation to fail if no binaries are available. Obviously this is expected behavior but perhaps we allow users to pass in flags as an alternative?

https://github.com/glotzerlab/workflows/blob/4510a18427564f3b1d656ac6abcdcd654912016e/setup-uv/action.yaml#L23

### PR DESCRIPTION
**Pull request by janbridley**
To resolve #8, the `--only-binary flag` has been made optional. While on by default (to ease in debugging), it can be disabled for packages with unresolvable dependencies.

### CODE REVIEW

File: setup-uv/action.yaml [hunk 1]
```diff
@@ -20,5 +24,7 @@ runs:
 
   - name: Install
     if: ${{ inputs.lockfile != '' }}
-    run: "uv pip sync ${{ inputs.lockfile }} --only-binary :all: --system --reinstall"
+    env:
+      ONLY_BINARY: ${{ {{ inputs.onlybinary }} && "--only-binary=:all:" || ""}}
```
Comment (janbridley) [inline setup-uv/action.yaml]:
```suggestion
      ONLY_BINARY: ${{ ${{ inputs.onlybinary }} && "--only-binary=:all:" || ""}}
```

File: setup-uv/action.yaml [hunk 2]
```diff
@@ -20,5 +24,7 @@ runs:
 
   - name: Install
     if: ${{ inputs.lockfile != '' }}
-    run: "uv pip sync ${{ inputs.lockfile }} --only-binary :all: --system --reinstall"
+    env:
+      ONLY_BINARY: ${{ ${{ inputs.onlybinary }} && "--only-binary=:all:" || ""}}
```
Comment (janbridley) [inline setup-uv/action.yaml]:
```suggestion
      ONLY_BINARY: ${{ inputs.onlybinary && "--only-binary=:all:" || ""}}
```

**Review by joaander — COMMENTED** [review]
Thanks! This looks good. Let's add a unit test to `test.yaml` that sets `onlybinary` to `'false'`.

**Comment by janbridley** [comment]
For some reason `${{ input.onlybinary }}` always evaluates to true, even if you pass in a typed (?) boolean. `${{ input.onlybinary==true }}` ~evaluates to false when expected~ always evaluates to false. 

This also appears to be the case for [402330](https://github.com/glotzerlab/workflows/pull/9/commits/402330a07207b4a363aafc3224793d619bbf38a3) where we compare strings. I am unsure how to proceed from here, I will read up on javascript.

**Comment by joaander** [comment]
Let's take a different approach then.

I modified the code to accept the input `only-binary` (with default ":all:") and pass the input directly to the command line `--only-binary${{ inputs.only-binary }}`. Callers can now use:
```
with:
  only-binary: ":none:"
```
to accept source packages. This is working for coxeter in https://github.com/glotzerlab/coxeter/actions/runs/9891585989/job/27322341562?pr=231

**Comment by janbridley** [comment]
> Let's take a different approach then.
> 
> I modified the code to accept the input `only-binary` (with default ":all:") and pass the input directly to the command line `--only-binary${{ inputs.only-binary }}`. Callers can now use:
> ...

Looks great! Thank you for that

**Pull request merged**